From 8ee332a396e5d7394d2d6b7897d1c0e5c67bb2d1 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 27 Nov 2023 13:46:10 +0800
Subject: [PATCH 9/9] support no initramfs situation

While running ostree without initramfs, it runs init program /init-ostree-pre.sh

Upstream-Status: Inappropriate [wrlinux specific]

Rebased to ostree 2023.7

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 src/switchroot/ostree-prepare-root-static.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/switchroot/ostree-prepare-root-static.c b/src/switchroot/ostree-prepare-root-static.c
index 4aaa469..2f63b90 100644
--- a/src/switchroot/ostree-prepare-root-static.c
+++ b/src/switchroot/ostree-prepare-root-static.c
@@ -388,6 +388,6 @@ main (int argc, char *argv[])
   if (mount ("none", "sysroot", NULL, MS_PRIVATE | MS_SILENT, NULL) < 0)
     err (EXIT_FAILURE, "remounting 'sysroot' private");
 
-  execl ("/sbin/init", "/sbin/init", NULL);
+  execl ("/init-ostree-pre.sh", "/init-ostree-pre.sh", deploy_path, NULL);
   err (EXIT_FAILURE, "failed to exec init inside ostree");
 }
-- 
2.27.0

